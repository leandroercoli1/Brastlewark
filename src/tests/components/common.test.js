import React from "react";
import renderer from "react-test-renderer";
import Alert from "components/common/alert";
import Loader from "components/common/loader";
import GithubBadge from "components/common/github-badge";
import UserBadge from "components/common/user-badge";

describe("renders common components correctly", () => {
  it("renders alert correctly", () => {
    const alert = renderer.create(<Alert>Alert message text</Alert>).toJSON();
    expect(alert).toMatchSnapshot();
  });
  it("renders loader correctly", () => {
    const loader = renderer.create(<Loader />).toJSON();
    expect(loader).toMatchSnapshot();
  });
  it("renders GitHub badge correctly", () => {
    const badge = renderer.create(<GithubBadge />).toJSON();
    expect(badge).toMatchSnapshot();
  });
  it("renders user badge correctly", () => {
    const badge = renderer.create(<UserBadge />).toJSON();
    expect(badge).toMatchSnapshot();
  });
});
