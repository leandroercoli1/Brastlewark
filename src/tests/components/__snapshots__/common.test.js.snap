// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders common components correctly renders GitHub badge correctly 1`] = `
<div
  className="github-badge"
>
  <a
    aria-current="true"
    href="https://github.com/leandroercoli/AxaTest"
    rel="noreferrer"
    target="_blank"
  >
    <img
      alt="github-icon"
      src="github-icon.png"
    />
  </a>
</div>
`;

exports[`renders common components correctly renders alert correctly 1`] = `
<div
  className="message-container message-fail"
>
  Alert message text
</div>
`;

exports[`renders common components correctly renders loader correctly 1`] = `
<div
  className="loader-container"
>
  <div
    className="loader-bar-wrapper"
  >
    <div
      className="loader-bar"
    />
  </div>
</div>
`;

exports[`renders common components correctly renders user badge correctly 1`] = `
<div
  className="user-badge"
  data-testid="user-badge"
>
  <img
    alt="logo"
    src="hero.jpg"
  />
</div>
`;
